From f04ccd0dcd81e2319c074098605a0e05b15d9818 Mon Sep 17 00:00:00 2001
From: Dirk Mueller <dirk@dmllr.de>
Date: Sat, 2 Feb 2019 09:19:45 +0100
Subject: [PATCH 2/2] ZipCheck: Also ignore RuntimeError

When there is a unpack error, we need to catch it. Potential
reasons are : NotImplementedError (which happens on compression
type issues) or RuntimeError (e.g for encrypted zip files).

This issue is already fixed in master in a non-backportable
way, so a new fix has been done on stable.
---
 ZipCheck.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ZipCheck.py b/ZipCheck.py
index a6f927a..4da6b06 100644
--- a/ZipCheck.py
+++ b/ZipCheck.py
@@ -43,7 +43,7 @@ class ZipCheck(AbstractCheck.AbstractCheck):
                     badcrc = z.testzip()
                     if badcrc:
                         printError(pkg, 'bad-crc-in-zip', badcrc, fname)
-                except zipfile.error:
+                except (RuntimeError, zipfile.error):
                     printWarning(pkg, 'unable-to-read-zip', '%s: %s' %
                                  (fname, sys.exc_info()[1]))
                 else:
-- 
2.11.0

